stdlib/Distributed/src/managers.jl

```diff
@@ -225,6 +268,68 @@ function launch_on_machine(manager::SSHManager, machine, cnt, params, launched,
 end
 
 
+function print_shell_escaped_windows(io, args::AbstractString...)::Nothing
```

Comment:
I have now also implemented an actual `cmd.exe` escaping algorithm on top of C-runtime ARGS escaping. The method I finally found (after several attempts) may seem a bit odd, but seems to work well: inserts a `^` escape character before any potential `cmd.exe` metacharacter whenever there was so far an even number of double quotes. This algorithms is needed because `cmd.exe` only half-treats double quotes as metacharacters: it does not remove them, but it uses them to toggle a quoting-mode flag, which in turn activates or deactivates other metacharacters, including `^`.

Reply:
I have tested this patch for far only in the case where a Linux leader machine invokes a worker on a Windows machine (my use case is remotely controlling from Linux some laboratory kit for which only a Windows driver DLL is available). I have not yet tested remote method calls from Windows to Windows or form Windows to Linux, but I wouldn't expect there any changes needed. In particular, I would not expect any need for setting `windows_verbatim=true`. Without this flag, libuv already applies locally the same C-runtime escaping algorithm to the ssh commandline that I also had implemented for the remote command line, and that is needed to pass remotecmd on to the ssh process on Windows.

Reply:
I've followed the function signature and naming of shell_escape_posixly() closely in case someone wants to put this into shell.jl, but I don't have particular views on what it should be called or where it should go.

Reply:
I think that's the clearest description of the cmd.exe algorithm I've ever heard.

I think it might be better to split this into a `shell_escape_win32ly` and `shell_escape_dosly` pair though, since some uses might only want one or the other (well, I don't care much about the exact names, just something memorable enough). I also think these are appropriate utility functions for inclusion in shell.jl. Can you move this there and add some tests? (you can probably just add them alongside the `shell_escape_posixly` tests)

Reply:
According to my (potentially flawed) understanding, all the following tests should pass with this function but not all do
```julia
   @test Base.shell_escape_windows("a b c", "d", "e") == "\"a b c\" d e"
   @test Base.shell_escape_windows("ab\"c", "\\", "d") == "ab\\\"c\" \"\\\\\" d"
   @test Base.shell_escape_windows("a\\\\b", "d\"e",  "f\"g", "h") == "a\\\\\\b d\"e f\"g h"
   @test Base.shell_escape_windows("a\"b", "c", "d") == "a\\\"b c d"
   @test Base.shell_escape_windows("a\\\\b c", "d", "e") == "a\\\\\\\\\"b c\" d e"
```
(from https://docs.microsoft.com/en-us/cpp/c-language/parsing-c-command-line-arguments?view=vs-2019)